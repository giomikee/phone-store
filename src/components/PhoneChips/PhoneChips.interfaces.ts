import type { SxProps, Theme } from '@mui/material';

export interface Props {
    brand: string;
    basePrice: number;
    brandChipStyle?: SxProps<Theme>;
}