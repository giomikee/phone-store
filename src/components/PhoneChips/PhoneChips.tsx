import { Box, Chip } from '@mui/material';
import EuroIcon from '@mui/icons-material/Euro';
import SmartphoneIcon from '@mui/icons-material/Smartphone';
import type { Props } from './PhoneChips.interfaces';
import { SIZES } from '../../constants';
import { formatPrice } from '../../utils';

const PhoneChips = ({brand, basePrice, brandChipStyle}: Props) => {
    return (
        <Box>
            <Chip
                color='primary'
                icon={<SmartphoneIcon />}
                label={brand}
                size={SIZES.small}
                sx={brandChipStyle}
            />
            <Chip
                color='secondary'
                icon={<EuroIcon />}
                label={`desde ${formatPrice(basePrice)}`}
                size={SIZES.small}
            />
        </Box>
    )
}

export default PhoneChips