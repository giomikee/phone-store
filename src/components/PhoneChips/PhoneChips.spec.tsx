import { describe, expect, it } from 'vitest';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/vitest';
import PhoneChips from './PhoneChips';

describe('Given PhoneChips component', () => { 
    const defaultProps = {
        brand: 'Apple',
        basePrice: 999,
    };

    describe('When component is rendered', () => { 
        it('Then it should match the snapshot', () => {
            const { container } = render(<PhoneChips {...defaultProps} />);
            expect(container).toMatchSnapshot();
        });
    });
});