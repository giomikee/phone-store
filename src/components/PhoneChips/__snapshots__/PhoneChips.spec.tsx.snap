// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Given PhoneChips component > When component is rendered > Then it should match the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-0"
  >
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorPrimary MuiChip-filledPrimary css-2mz69a-MuiChip-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorPrimary css-1umw9bq-MuiSvgIcon-root"
        data-testid="SmartphoneIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 1.01 7 1c-1.1 0-2 .9-2 2v18c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V3c0-1.1-.9-1.99-2-1.99M17 19H7V5h10z"
        />
      </svg>
      <span
        class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
      >
        Apple
      </span>
    </div>
    <div
      class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSecondary MuiChip-filledSecondary css-3p8epo-MuiChip-root"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiChip-icon MuiChip-iconSmall MuiChip-iconColorSecondary css-1umw9bq-MuiSvgIcon-root"
        data-testid="EuroIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M15 18.5c-2.51 0-4.68-1.42-5.76-3.5H15l1-2H8.58c-.05-.33-.08-.66-.08-1s.03-.67.08-1H15l1-2H9.24C10.32 6.92 12.5 5.5 15 5.5c1.61 0 3.09.59 4.23 1.57L21 5.3C19.41 3.87 17.3 3 15 3c-3.92 0-7.24 2.51-8.48 6H3l-1 2h4.06c-.04.33-.06.66-.06 1s.02.67.06 1H3l-1 2h4.52c1.24 3.49 4.56 6 8.48 6 2.31 0 4.41-.87 6-2.3l-1.78-1.77c-1.13.98-2.6 1.57-4.22 1.57"
        />
      </svg>
      <span
        class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
      >
        desde 999.00 â‚¬
      </span>
    </div>
  </div>
</div>
`;
